### PR TITLE
Add attribute of raw string to Scalar String_

### LINKED ISSUE #576: Get unparsed string from String_ for escaped sequences
I'm currently using PHPStan, which uses PHP7 parser provided by `PHP-Parser`.

My objective here is to get unparsed string argument from `String_` class such that I'll be able to detect if there's any escaped sequence in the string.

e.g. I wish to get argument of value `'This is a string with escape sequence \x41'` from `String_` class after parsing the code below.
```php
<?php
Foo::bar("This is a string with escape sequence \x41");
```
However, what I get instead is 
```
'This is a string with escape sequence A'
```
This is because the raw string was parsed.
I wonder if it is possible to get an unparsed string from `String_` class?

### PR DESCRIPTION
**Pull request by Billz95**
Fixes #576

### CODE REVIEW

File: lib/PhpParser/Node/Scalar/String_.php [hunk 1]
```diff
@@ -16,6 +16,9 @@ class String_ extends Scalar
     /** @var string String value */
     public $value;
 
+    /** @var string Unparsed String value */
+    public $rawString;
```
Comment (TomasVotruba) [inline lib/PhpParser/Node/Scalar/String_.php]: `$rawValue` would be better, to be consistent with `$value` above

File: lib/PhpParser/Parser/Php7.php [hunk 2]
```diff
@@ -2173,6 +2173,7 @@ protected function initReduceCallbacks() {
             433 => function ($stackPos) {
                  $attrs = $this->startAttributeStack[$stackPos-(1-1)] + $this->endAttributes; $attrs['kind'] = ($this->semStack[$stackPos-(1-1)][0] === "'" || ($this->semStack[$stackPos-(1-1)][1] === "'" && ($this->semStack[$stackPos-(1-1)][0] === 'b' || $this->semStack[$stackPos-(1-1)][0] === 'B')) ? Scalar\String_::KIND_SINGLE_QUOTED : Scalar\String_::KIND_DOUBLE_QUOTED);
             $this->semValue = new Scalar\String_(Scalar\String_::parse($this->semStack[$stackPos-(1-1)]), $attrs);
+            $this->semValue->rawString = $this->semStack[$stackPos-(1-1)];
```
Comment (localheinz) [inline lib/PhpParser/Parser/Php7.php]: What do you think about adjusting the indentation here?

Looks like both lines are incorrectly indented.

**Comment by Billz95** [comment]
Thanks for the reviews, I've updated my code accordingly and squashed them to one commit :sunny:

**Comment by othercorey** [comment]
@nikic Is this something you plan to expect to approve?

**Comment by TomasVotruba** [comment]
Similar node would be useful for `LNumber` and `DNumber`. Now it's impossible to separate `1_000` and `1000` and we have to work with tokens in Rector.

**Comment by TomasVotruba** [comment]
@nikic If you plan to accept this, I can submit PR with rebasd and correct implementation in grammer recipe.

**Comment by othercorey** [comment]
@TomasVotruba Might be better to just create a new PR. He doesn't seem to respond to some PRs.

**Comment by nikic** [comment]
Closing in favor of #831, which implements this as an attribute.

**Pull request closed without merging**